api/users/serializers.py

```diff
@@ -128,20 +130,28 @@ def profile_image_url(self, user):
         size = self.context['request'].query_params.get('profile_image_size')
         return user.profile_image_url(size=size)
 
+    def validate_social(self, value):
+        schema = from_json('social-schema.json')
+        try:
+            jsonschema.validate(value, schema)
+        except jsonschema.ValidationError as e:
+            raise InvalidModelValueError(e)
+
+        social_dict = {}
+        for key, val in value.iteritems():
+            # Ignore fields that are not specified in the current social fields list
```

Comment:
Just for history's sake, when I wrote this the requirement was "Ignore attempts to update social fields that are not in [the list of fields]" which is what this was doing. Requirement has changed so will now error with extra fields instead of ignoring.